libs/go/libp2p_node/dht/dhtpeer/dhtpeer.go

```diff
@@ -348,9 +349,10 @@ func New(opts ...Option) (*DHTPeer, error) {
 	// if peer is joining an existing network, announce my agent address if set
 	if len(dhtPeer.bootstrapPeers) > 0 {
 		// TOFIX, set to true even when no address present
-		dhtPeer.addressAnnounced = true
-		ldebug().Msg("Address was announced on bootstrap peers")
-		if dhtPeer.myAgentAddress != "" {
+		dhtPeer.enableAddressAnnouncement = true
```

Comment:
Can you explain?

Reply:
during developement I found if no connections to other peers it's not possible to announce address, so initial idea of flag is check is possible to announce or not.

Reply:
This comment could be left in code for posterity